enaml/core/code_generator.py

```diff
@@ -308,27 +323,49 @@ def load_build_class(self):
             bc.Instr("LOAD_BUILD_CLASS"),               # TOS -> builtins.__build_class__
         )
 
-    def make_function(self, n_defaults=0):
+    def make_function(self, n_defaults=0, name=None):
         """ Make a function from a code object on the TOS.
 
         """
+        if not PY311:
+            self.load_const(name)
         self.code_ops.append(                           # TOS -> qual_name -> code -> defaults
             bc.Instr("MAKE_FUNCTION", n_defaults),      # TOS -> func
```

Comment:
Under 3.11 MAKE_FUNCTION does not take the number of default anymore but a code (see https://docs.python.org/3.11/library/dis.html#opcode-MAKE_FUNCTION). This probably explain your issue.

Reply:
It looks like this has been the case ever since 2.7, anyways there is only one call to make_function with defaults and it happens to be `0x01` so that is why it worked.

Reply:
I thought it could be the origin of your default argument issue. I will have to look elsewhere (possibly in the C helpers)